packages/hash/shared/src/save.ts

```diff
@@ -312,11 +316,61 @@ const updateBlocks = defineOperation(
                 },
               });
             }
+          } else {
+            const draftBlock = getDraftEntityFromEntityId(
+              entityStore.draft,
+              savedEntity.entityId,
+            );
+            const draftChildEntityId = draftBlock?.properties.entity?.entityId;
+            const draftChildEntityWithDraftId = getDraftEntityFromEntityId(
+              entityStore.draft,
+              draftChildEntityId,
+            );
+
+            if (draftChildEntityWithDraftId && draftChildEntityId) {
+              const { draftId: _, ...draftChildEntity } =
+                draftChildEntityWithDraftId;
+
+              // Check if a block's data has been updated
+              if (
+                draftChildEntity &&
+                !isEqual(
+                  draftChildEntity.properties,
+                  savedChildEntity.properties,
+                )
+              ) {
+                // If the block's data was updated and the entityId changed
+                // then the block data was swapped. Push swapBlockData action
+                if (draftChildEntity.entityId !== savedChildEntity.entityId) {
+                  updates.push({
+                    swapBlockData: {
+                      entityId: savedEntity.entityId,
+                      accountId: savedEntity.accountId,
+                      newEntityAccountId: draftChildEntity.accountId!,
+                      newEntityEntityId: draftChildEntity.entityId!,
+                    },
+                  });
+                } else {
+                  // If the block's data was updated and the entityId
+                  // didn't change, push an updateEntity action
+                  // @todo push an updateEntity action
```

Comment:
Initially tried to push an` updateEntity` action here

```
updates.push({
        updateEntity: {
              entityId: savedChildEntity.entityId,
                accountId: savedChildEntity.accountId,
                properties: draftChildEntity.properties,
          },
});
```
but that didn't seem to fix the reset issue that happens when a block's data is updated via BlockConfigMenu.